src/app/editor/code/editor-container/codegen/xamarin-forms/xamarin-forms.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import { CodeGenFacade, XlayersNgxEditorModel } from '../codegen.service';
+import { XamarinFormsCodeGenVisitor } from './codegen/xamarin-forms-codegenvisitor.service';
+import { readmeTemplate, mainPageTemplate } from './xamarin-forms.template';
+
+/**
+ * @see CodeGenFacade implementation able to generate Xamarin.Forms code
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class XamarinFormsCodeGenService implements CodeGenFacade {
+
+  constructor(private readonly codegen: XamarinFormsCodeGenVisitor) {}
+
+  buttons() {
+    return {
+      stackblitz: false
+    };
+  }
+
+  generate(ast: SketchMSLayer): Array<XlayersNgxEditorModel> {
+    this.codegen.fileList = [];
```

Comment:
`fileList` should not be initialized from outside the class. And being initialized inside the class like :
```typescript
fileList: Array<XlayersNgxEditorModel> = [];
```
And concat the given template result:
```typescript
[
  {
    uri: "README.md",
    value: this.generateReadme(),
    language: "markdown",
    kind: "text"
  },
  {
    uri: "MainPage.xaml",
    value: this.generateComponent(ast),
    language: "xaml",
      kind: "xamarinForms"
  },
  ...this.codegen.fileList
];
```

Reply:
This is more a reset than an initializer.
It is required to prevent files from being added to the list multiple times when the user change his desired language

Reply:
The fileList side effect associated with this `unshift` is really confusing.

Merging `XamarinFormsCodeGenVisitor` and `XamarinFormsCodeGenGenService` solves the problem since the files `README.md`, `MainPage.xaml` and dynamic files like SVG will be generated in the same place. So the `fileList` can be reset at the end of the generation.